optimum/habana/transformers/models/glm4v/modeling_chatglm.py

```diff
@@ -0,0 +1,1787 @@
+                        # WA for long sequence, better perf. than recompute
+                        if (q_len > 16384 or (q_len >= 6144 and bsz >= 2)) and self.training:
+                            attn_output = self.gaudi_flash_attn_v1(
+                                query_layer, key_layer, value_layer, attention_mask, dropout_rate, softmax_mode
+                            )
+                        else:
+                            attn_output = self.fused_scaled_dot_product_attention(
+                                query_layer,
+                                key_layer,
+                                value_layer,
+                                attention_mask,
+                                dropout_rate,
+                                False,
+                                None,
+                                softmax_mode,
+                            )
+        else:
+            query_layer, key_layer, value_layer, attention_mask = gaudi_chatglm_repeat_kv(
+                query_layer, key_layer, value_layer, attention_mask
+            )
+            attn_weights = self.matmul_qk(query_layer, key_layer.transpose(-2, -1)) / self.norm_factor
+
+            if self.coeff is not None:
+                attn_weights = attn_weights * self.coeff
+
+            if attention_mask is not None:  # no matter the length, we just slice it
+                causal_mask = attention_mask
+                if cache_position is not None:
+                    causal_mask = attention_mask[:, :, cache_position, : key_layer.shape[-2]]
+                attn_weights = attn_weights + causal_mask
+
+            if attn_softmax_bf16:
+                attn_weights = torch.nn.functional.softmax(attn_weights, dim=-1, dtype=query_layer.dtype)
+            else:
+                # upcast attention to fp32
+                attn_weights = torch.nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32).to(
+                    query_layer.dtype
+                )
+            if self.training:
+                attn_weights = self.attention_dropout(attn_weights)
+            attn_output = self.matmul_av(attn_weights, value_layer)
+            attn_output = attn_output.reshape(bsz, -1, q_len, self.hidden_size_per_attention_head)
+
+        # =================
+        # Output. [sq, b, h]
+        # =================
+
+        attn_output = attn_output.permute(2, 0, 1, 3).contiguous()
+
+        context_layer = attn_output.reshape(q_len, bsz, self.hidden_size_per_partition)
+
+        return context_layer
+
+
+CORE_ATTENTION_CLASSES = {"eager": CoreAttention, "sdpa": CoreAttention, "flash_attention_2": CoreAttention}
+
+
+class SelfAttention(torch.nn.Module):
+    """Parallel self-attention layer abstract class.
+
+    Self-attention layer takes input with size [s, b, h]
+    and returns output of the same size.
+    """
+
+    def __init__(self, config: GLM4VConfig, layer_number, device=None):
+        super(SelfAttention, self).__init__()
```

Comment:
```suggestion
        super().__init__()
```

In Python 3, using super().__init__() is generally preferred